master/internal/api_workspace.go

```diff
@@ -56,19 +60,45 @@ func maskStorageConfigSecrets(w *workspacev1.Workspace) error {
 	return nil
 }
 
+func validatePostWorkspaceRequest(req *apiv1.PostWorkspaceRequest) error {
+	if (req.ClusterName != nil || req.Quota != nil) &&
+		(req.NamespaceName == nil && !req.AutoCreateNamespace) {
+		return status.Errorf(codes.InvalidArgument,
+			"Must specify either an existing Kubernetes namespace or indicate that you would like a namespace to be auto-created")
+	}
+	if req.NamespaceName != nil && req.Quota != nil {
+		return status.Errorf(codes.InvalidArgument,
+			"You cannot set a quota on an already existing k8s namespace.")
+	}
+	if (req.AutoCreateNamespace || req.NamespaceName != nil) && req.ClusterName == nil {
+		return status.Errorf(codes.InvalidArgument,
+			"You must specify a cluster for the specified namespace that you would like to bind.")
+	}
+	return nil
+}
+
 func validateWorkspaceName(name string) error {
 	switch {
 	case len(name) < 1:
 		return status.Errorf(codes.InvalidArgument, "name '%s' must be at least 1 character long", name)
-	case len(name) > 80:
-		return status.Errorf(codes.InvalidArgument, "name '%s' must be at most 80 character long", name)
+	case len(name) > 53:
```

Comment:
why is there a 53 char limit?

Reply:
I was leaving room for the prefix and suffix for an auto-created namespace (since Kubernetes namespaces cannot be longer than 63 characters). However, I removed the suffix, so now an auto-generated namespace would simply follow the convention of `det-<workspace_name>`. this leaves room for a workspace name of at most 59 characters, so I'll change this!